### PR TITLE
feat: Resolve most front-end warnings

### LINKED ISSUE #1068: Deal with front end warnings
Alot of warnings are given by React due to things like missing dependencies in `useEffect`

Errors to solve:

./src/components/manageProjects/editMeetingTimes.js
[1] Line 20:9: 'URL' is assigned a value but never used no-unused-vars
[1] Line 23:10: 'eventToEdit' is assigned a value but never used no-unused-vars
[1] Line 29:6: React Hook useEffect has missing dependencies: 'projectToEdit._id' and 'recurringEvents'. Either include them or remove the dependency array. If 'setREvents' needs the current value of 'recurringEvents', you can also switch to useReducer instead of useState and read 'recurringEvents' in the reducer react-hooks/exhaustive-deps
[1]
[1] ./src/components/auth/HandleAuth.js
[1] Line 24:6: React Hook useEffect has a missing dependency: 'props.location.search'. Either include it or remove the dependency array react-hooks/exhaustive-deps
[1] Line 32:5: React Hook useEffect has a missing dependency: 'auth.isError'. Either include it or remove the dependency array react-hooks/exhaustive-deps
[1] Line 44:7: React Hook useEffect has a missing dependency: 'auth'. Either include it or remove the dependency array react-hooks/exhaustive-deps
[1] Line 44:26: React Hook useEffect has a complex expression in the dependency array. Extract it to a separate variable so it can be statically checked react-hooks/exhaustive-deps
[1]
[1] ./src/components/admin/dashboard/index.js
[1] Line 331:6: React Hook useEffect has a missing dependency: 'getAndSetData'. Either include it or remove the dependency array react-hooks/exhaustive-deps
[1]
[1] ./src/components/manageProjects/utilities/addDurationToTime.js
[1] Line 33:4: Expected an error object to be thrown no-throw-literal
[1]
[1] ./src/components/manageProjects/editableField.js
[1] Line 34:6: React Hook useEffect has missing dependencies: 'checkUser' and 'fieldData'. Either include them or remove the dependency array. If 'setFieldValue' needs the current value of 'fieldData', you can also switch to useReducer instead of useState and read 'fieldData' in the reducer react-hooks/exhaustive-deps
[1]
[1] ./src/pages/Events.js
[1] Line 36:8: React Hook useEffect has a missing dependency: 'fetchData'. Either include it or remove the dependency array react-hooks/exhaustive-deps
[1]
[1] ./src/components/ReadyEvents.js
[1] Line 37:8: React Hook useEffect has a missing dependency: 'fetchEvent'. Either include it or remove the dependency array react-hooks/exhaustive-deps
[1]
[1] ./src/pages/Success.js
[1] Line 42:11: Emojis should be wrapped in , have role="img", and have an accessible description with aria-label or aria-labelledby jsx-a11y/accessible-emoji
[1] Line 43:11: Emojis should be wrapped in , have role="img", and have an accessible description with aria-label or aria-labelledby jsx-a11y/accessible-emoji
[1] Line 44:11: Emojis should be wrapped in , have role="img", and have an accessible description with aria-label or aria-labelledby jsx-a11y/accessible-emoji
[1]
[1] ./src/pages/ProjectLeaderDashboard.js
[1] Line 42:3: The 'getNextEvent' function makes the dependencies of useEffect Hook (at line 257) change on every render. Move it inside the useEffect callback. Alternatively, wrap the 'getNextEvent' definition into its own useCallback() Hook react-hooks/exhaustive-deps
[1] Line 62:3: The 'getRoster' function makes the dependencies of useEffect Hook (at line 253) change on every render. Move it inside the useEffect callback. Alternatively, wrap the 'getRoster' definition into its own useCallback() Hook react-hooks/exhaustive-deps
[1] Line 90:3: The 'getAttendees' function makes the dependencies of useEffect Hook (at line 249) change on every render. Move it inside the useEffect callback. Alternatively, wrap the 'getAttendees' definition into its own useCallback() Hook react-hooks/exhaustive-deps
[1] Line 130:3: The 'getDashboardInfo' function makes the dependencies of useEffect Hook (at line 245) change on every render. Move it inside the useEffect callback. Alternatively, wrap the 'getDashboardInfo' definition into its own useCallback() Hook react-hooks/exhaustive-deps
[1]
[1] ./src/components/dashboard/AttendeeTable.js
[1] Line 42:41: Expected to return a value in arrow function array-callback-return
[1] Line 62:8: React Hook useEffect has a missing dependency: 'sortAttendees'. Either include it or remove the dependency array react-hooks/exhaustive-deps
[1]
[1] ./src/pages/Home.js
[1] Line 45:8: React Hook useEffect has a missing dependency: 'fetchEvents'. Either include it or remove the dependency array react-hooks/exhaustive-deps
[1]
[1] ./src/pages/ManageProjects.js
[1] Line 55:6: React Hook useEffect has missing dependencies: 'fetchProjects' and 'fetchRecurringEvents'. Either include them or remove the dependency array react-hooks/exhaustive-deps
[1] Line 102:7: Unreachable code no-unreachable
[1] Line 114:7: Unreachable code no-unreachable
[1]
[1] ./src/components/dashboard/RosterTable.js
[1] Line 56:13: 'email' is assigned to itself no-self-assign
[1] Line 57:14: 'fileId' is assigned to itself no-self-assign
[1] Line 84:9: 'gitHubClickHandler' is assigned a value but never used no-unused-vars
[1]
[1] ./src/pages/CheckInForm.js
[1] Line 582:4: React Hook useEffect has a missing dependency: 'fetchQuestions'. Either include it or remove the dependency array react-hooks/exhaustive-deps
[1]
[1] ./src/components/admin/donutChartContainer.js
[1] Line 75:6: React Hook useEffect has missing dependencies: 'createArc', 'createPie', and 'pieData'. Either include them or remove the dependency array react-hooks/exhaustive-deps
[1]
[1] ./src/pages/Event.js
[1] Line 92:8: React Hook useEffect has a missing dependency: 'fetchEvent'. Either include it or remove the dependency array react-hooks/exhaustive-deps

### PR DESCRIPTION
**Pull request by trillium**
Fixes #1068 

### What changes did you make and why did you make them ?

Front end compiles with many warnings, this removes all but one of them at time of writing.

A few of the resolutions are //disable-next-line while we figure out if we want to drop some of the older features.

## Changes

<details>
<summary>Squashed Commit Messages:</summary>

asset: Update caniuse

fix: Remove useEffect from Auth, add onSubmit

Resolve Warning:
* Line 104:6:  React Hook useEffect has a missing dependency: 'handleLogin'. Either include it or remove the dependency array  react-hooks/exhaustive-deps

fix: Add props.history to useEffect dep array

* Resolves Line 13:8:  React Hook useEffect has a missing dependency: 'props.history'. Either include it or remove the dependency array  react-hooks/exhaustive-deps

fix: Add span and aria-label to eomji

fix: Remove useEffect, resolve no-unused-vars

in components:
* UserDashboard
* UserProfile
* ReturningUser
* Auth

fix: Add eslint ignore to recrutingDataFormatted

Code it references is currently commented out
Preserve until decision is made on to remove it or not

fix: Resolve no-self-assign warning in RosterTable

fix: Add ignore rule for gitHubClickHandler

fix: Remove unused vars event_city and event_state

fix: Add ignore rule for setCheckInReady

fix: Change .map to .forEach, no return needed

fix: Resolve no-throw-literal using new Error()

fix: Comment out console.log for backend

fix: Move true/false to prop vs new component

fix: Refactor single use fetch inside useEffect

Move fetch functions into useEffect to resolve react-hooks/exhaustive-deps

fix: Add props.location.search to deps array

Resolve react-hooks/exhaustive-deps

fix: Add auth to useEffect deps array

fix: Move fetch inside useEffect for eslint

* Add props to deps array to resolve react-hook/exhaustive-deps Resolves react-hooks/exhaustive-deps

fix: Move fetch inside useEffect for eslint

fix: Move fetch inside useEffect for eslint

fix: Rename headersToSend in import

Resolves warning for deps array, headersToSend is now defined outside the scope of the react component.

fix: Rename headersToSend in import

fix: Rename headersToSend in import

fix: Rename headersToSend in import

fix: Rename headersToSend in import

fix: Rename headersToSend in import

fix: Rename headersToSend in import

fix: Update useEffect react-hooks/exhastive-deps

Move single use functions inside useEffect
Add pieData to deps array

fix: Move single use function inside useEffect

Resolves react-hooks/ehxaustive-deps

</details>

<details>
<summary>Changes Before:</summary>
image

</details>

<details>
<summary>Changes After:</summary>
image
</details>

### CODE REVIEW

File: backend/workers/createRecurringEvents.js [hunk 1]
```diff
@@ -140,26 +140,26 @@ module.exports = (cron, fetch) => {
...
     async function runTask() {
-        console.log("Creating today's events");
+        // console.log("Creating today's events");
 
         await fetchEvents();
         await fetchRecurringEvents();
         await filterAndCreateEvents();
 
-        console.log("Today's events are created");
+        // console.log("Today's events are created");
```
Comment (MattPereira) [inline backend/workers/createRecurringEvents.js:162]: My preference would be to outright remove all of the `console.log()`s to keep things a bit more tidy but I'm also open to discussion if there is a reason we should keep them as comments

**Review by MattPereira — CHANGES_REQUESTED** [review]

**Comment by MattPereira** [comment]
@Spiteless 

Really appreciate you grinding down so many of these errors. Love what you did using the import alias `as headerToSend` to DRY up the code

I don't think any of the features of the app are broken by this PR, but we should still have another team member look at this for safety since there are so many files changed

Moving forward my preference would be to decompose issues and PRs into smaller chunks to make it easier to confidently review the changes and be sure nothing is breaking

I'm willing to invest in someone like you who is clearly committed to grinding, but I also want to kindly suggest that you might consider also joining the [hackforla/website](https://github.com/hackforla/website) team to learn about the kanban system that Bonnie has spent the last few years perfecting. The project does an amazing job teaching devs the fundamentals of github collaboration

**Comment by trillium** [comment]
Yes, I tried getting the last one sorted out for about four days. I made
progress but not enough to resolve it completely. I had planned on asking
for help to get the last one resolved, but I’m not really sure it’s even
necessary because we might not use that component  at all within the
current iteration of VRMS.

Thanks for the vote of confidence on time investment. Would it have been
better to split this PR into similar chunks? For example, maybe 3 to 6 of
the resolutions to warnings were moving the definition of headersToSend
outside of the React component and another chunk was moving single use
fetch call within the usEffect hook rather than defining them outside the
hook.

In regard to joining the hackforla website team, I only have enough time
for one weekly evening meeting as it directly conflicts with my work
schedule (I work evenings and weekends). I am capable of following the
kanban system, what you’ve been seeing lately was me overcompensating for
not doing a  whole lot on the project prior to me getting my new work
computer.

If I’d be a better fit there that’s fine too, I’d like to get to the point
of being hireable so I can work less evenings :D

Anyhow, thanks for your focus and attention on my PRs, looking forward to
talking this evening.

**Comment by MattPereira** [comment]
@Spiteless 

Yeah you are spot on with your idea to split the error fixes into groups. Ideally, you would create a new issue for each group of errors you are trying to solve and make a PR for each issue. That way you would also rack up the Github stats for creating issues and having multiple PRs merged and it would also make it easier for PR reviewers to assess the situation and if something did go wrong we would more easily be able to discern specifically what code broke something with the live site.

**Comment by trillium** [comment]
Closing this PR to create smaller PRs with similar ESLint Warning resolutions

**Pull request closed without merging**